### ISSUE TITLE
Compile fails with 'error: format not a string literal and no format arguments'

### ISSUE THREAD

**Issue by merkys** [electronic-structure/SIRIUS#444]
Compiling `src/Band/band.cpp` to `CMakeFiles/sirius.dir/Band/band.cpp.o` fails with the following:
```
[ 13%] Building CXX object src/CMakeFiles/sirius.dir/Band/band.cpp.o
cd <builddir>/sirius/obj-x86_64-linux-gnu/src && /usr/bin/c++  -D__PROFILE -D__PROFILE_TIME -D__USE_MEMORY_POOL -D__USE_TIMER -I<builddir>/sirius/obj-x86_64-linux-gnu/src -I<builddir>/sirius/src/SDDK -I<builddir>/sirius/src -I<builddir>/sirius/src/. -isystem /usr/lib/x86_64-linux-gnu/openmpi/include -isystem /usr/lib/x86_64-linux-gnu/openmpi/include/openmpi -isystem /usr/include/hdf5/serial  -g -O2 -fdebug-prefix-map=<builddir>/sirius=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -fPIC -fvisibility=hidden   -fopenmp -std=gnu++14 -o CMakeFiles/sirius.dir/Band/band.cpp.o -c <builddir>/sirius/src/Band/band.cpp
In file included from <builddir>/sirius/src/K_point/k_point_set.hpp:28,
                 from <builddir>/sirius/src/Band/band.cpp:27:
<builddir>/sirius/src/K_point/k_point.hpp: In instantiation of ‘void sirius::K_point::message(int, const char*, Args ...) const [with Args = {const char*}]’:
<builddir>/sirius/src/Band/band.cpp:425:5:   required from ‘void sirius::Band::check_residuals(sirius::Hamiltonian_k&) const [with T = double]’
<builddir>/sirius/src/Band/band.cpp:414:9:   required from ‘void sirius::Band::initialize_subspace(sirius::Hamiltonian_k&, int) const [with T = double]’
<builddir>/sirius/src/Band/band.cpp:154:46:   required from here
<builddir>/sirius/src/K_point/k_point.hpp:746:19: error: format not a string literal and no format arguments [-Werror=format-security]
  746 |             printf(args...);
      |             ~~~~~~^~~~~~~~~
cc1plus: some warnings being treated as errors
```
Top of `cmake` output:
```
-- The C compiler identification is GNU 9.2.1
-- The CXX compiler identification is GNU 9.2.1
```

**Comment by simonpintarelli** [comment]
Thank you for reporting, we were not compiling with `-Werror=format-security` so far. It's fixed in PR #446 .

**Comment by merkys** [comment]
Thanks a lot for the prompt fix! Compile completes successfully now.
